src/main/java/com/greghaskins/spectrum/internal/exception/ExceptionExpectationProxy.java

```diff
@@ -0,0 +1,77 @@
+package com.greghaskins.spectrum.internal.exception;
+
+import com.greghaskins.spectrum.Block;
+import com.greghaskins.spectrum.ExceptionExpectation;
+import com.greghaskins.spectrum.internal.configuration.BlockConfiguration;
+import com.greghaskins.spectrum.internal.hooks.Hook;
+import com.greghaskins.spectrum.internal.hooks.NonReportingHook;
+
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+/**
+ * Proxy to the exception expectation supplier.
+ */
+public class ExceptionExpectationProxy implements ExceptionExpectation {
```

Comment:
If we merge the _proxy code_ I may be able to dispense with this proxy class... I think I'd like to do that.